pep-0522.txt

```diff
@@ -76,78 +125,84 @@ However, this guidance has also come with a longstanding caveat: developers
 writing security sensitive software at least for Linux, and potentially for
 some other \*BSD systems, may need to wait until the operating system's
 random number generator is ready before relying on it for security sensitive
-operations.
-
-Unfortunately, there's currently no clear indicator to developers that their
-software may not be working as expected when run early in the Linux boot
-process, or on hardware without good sources of entropy to seed the operating
-system's random number generator: due to the behaviour of the underlying
-``/dev/urandom`` device, ``os.urandom()`` on Linux returns a result either way,
-and it takes extensive statistical analysis to show that a security
-vulnerability exists.
+operations. This generally only occurs if ``os.urandom()`` is read very
+early in the system initialization process, or on systems with few sources of
+available entropy (e.g. some kinds of virtualized or embedded systems), but
+unfortunately the exact conditions that trigger this are difficult to predict,
+and when it occurs then there is no direct way for userspace to tell it has
+happened without querying operating system specific interfaces.
+
+On \*BSD systems, encountering this situation means ``os.urandom()`` will block
```

Comment:
This is a minor nitpick, but AFAICT (could easily be misinformed on one piece of another but):
* OpenBSD: requires entropy be present before spawning the first userspace process
* FreeBSD: reads from `/dev/urandom` and `/dev/random` block if the RNG is uninitialized, and never afterwards (as described here)
* NetBSD: acts like Linux

This is why my draft avoided saying anything about `*BSD` :-)